feat: match data points and labels to corresponding frame index

BrandonElk opened the pull request:
Adds frame index and label to dataframe

cosmo-k1 committed:
here is the preprocessor

outputs a dataframe given a video!!!!

cosmo-k1 committed:
update

cosmo-k1 committed:
got rid of unnecessary comments

cosmo-k1 committed:
update (useless function brandon already did)

oThommy committed:
build: add telemetry-parser to requirements

cosmo-k1 committed:
update denk ik

cosmo-k1 committed:
fixed linspace to arange

stupid numpy

cosmo-k1 committed:
update

BrandonElk committed:
Labeler first model

Adds frame index and label to dataframe

BrandonElk committed:
build: add av to requirements

oThommy commented on notebooks/labeler.ipynb:
```diff
@@ -0,0 +1,273 @@
+    "    dataframe = add_frame_index(dataframe, total_frames, frame_intervals)\n",
+    "    label_dict = sort_frametolabels(dataframe_labeled_frames)\n",
+    "    dataframe = match_labeltoframe(dataframe, label_dict)\n",
+    "    dataframe.dropna(inplace=True)\n",
```
```suggestion
    "    dataframe.dropna(inplace=True) # Remove data outside video duration (before the first or after the last frame)\n",
```

oThommy commented on notebooks/labeler.ipynb:
```diff
@@ -0,0 +1,273 @@
+    "# FUNCTION TO RUN EVERYTHING\n",
+    "\n",
+    "\n",
+    "def runner(video_path, dataframe, dataframe_labeled_frames):\n",
```
Tip: I notice you use a lot of functions and if that works for you, great! I just want to point out that you shouldn't feel obligated to always use functions. 

Personally, I view notebooks as exploratory experiments which are later refactored into `.py` files (e.g., in `src/` directory), so it's ok to do more '1d-programming'. In notebooks, I like to use functions for reusable logic but often skip them for quick, one-off tasks.

oThommy commented on notebooks/labeler.ipynb:
```diff
@@ -0,0 +1,273 @@
+    "\n",
+    "    # Decode video frames\n",
+    "    for i, frame in enumerate(container.decode(video=0)):\n",
+    "\n",
```
```suggestion
    "        # TODO: If the video stream is corrupted, it could cause infinitely repeated decoding attempts. Might need if check in case it's unstable during testing\n",
    "\n",
```

oThommy reviewed (changes requested):
@BrandonElk please go through the comments and resolve/commit them.

To merge the pull request after you are done, 

- select 'Squash and merge' from the dropdown menu (instead of 'Merge pull request'),
- leave the 'Commit header' as-is, and
- remove/clear all text under 'Commit message' (i.e., remove the text "Adds frame index and label to dataframe").

BrandonElk committed:
Update notebooks/labeler.ipynb

Co-authored-by: Thom van den Hil <78319191+oThommy@users.noreply.github.com>

cosmo-k1 committed:
made Raf's changes. much faster parser

BrandonElk committed:
Update notebooks/labeler.ipynb

Co-authored-by: Thom van den Hil <78319191+oThommy@users.noreply.github.com>

cosmo-k1 committed:
Feature/gpmf extraction preprocessing (#10)

Implemented Raf's changes, faster parser. The master function is in extraction_functions.py

oThommy reviewed (approved):
lgtm!
